lib/vrl/cli/src/repl.rs

```diff
@@ -182,10 +212,33 @@ impl Hinter for Repl {
     type Hint = String;
 
     fn hint(&self, line: &str, pos: usize, ctx: &Context<'_>) -> Option<String> {
-        self.hinter.hint(line, pos, ctx)
+        if pos < line.len() {
+            return None;
+        }
+
+        // Check history first
+        if let Some(hist) = self.history_hinter.hint(line, pos, ctx) {
+            return Some(hist);
+        }
+
+        // Then check the other built-in hints
+        self.hints
+            .iter()
+            .filter_map(|hint| {
+                if pos > 0 && hint.starts_with(&line[..pos]) {
+                    Some(suffix(hint, pos))
+                } else {
+                    None
+                }
+            })
+            .next()
     }
 }
 
+fn suffix(s: &str, pos: usize) -> String {
+    String::from(&s[pos..])
+}
```

Comment:
This function is only used once, so I don't see a need to extract this from the call-site.